cloud/amazon/ec2_vpc_peer.py

```diff
@@ -0,0 +1,1544 @@
+    params = {
+        'VpcId': vpc_id,
+        'PeerVpcId': vpc_peer_id,
+        'DryRun': check_mode
+    }
+    if peer_owner_id:
+        params['PeerOwnerId'] = peer_owner_id
+
+    success, err_msg, results = (
+        describe_peering_connections(
+            client, params['VpcId'], params['PeerVpcId'],
+            status_codes=['active']
+        )
+    )
+    if results:
+        updateit = True
+    else:
+        runit = True
+
+    if runit:
+        success, changed, err_msg, results = (
+            runner(client, 'present', params)
+        )
+        if success and changed:
+            updateit = True
+
+    if updateit:
+        if isinstance(results, list):
+            results = convert_to_lower(results[0])
+        vpc_peering_id = results['vpc_peering_connection_id']
+        success, changed, err_msg, results = (
+            update(
+                client, vpc_peering_id, tags, accept_peer,
+                accept_with_profile, region, accepter_routes,
+                requester_routes, check_mode
+            )
+        )
+        if success:
+            err_msg = ''
+
+    results = convert_to_lower(results)
+    if results.get('tags', None):
+        results['tags'] = convert_list_of_tags(results['tags'])
+
+    return success, changed, err_msg, results
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            accepter_routes=dict(type='list'),
+            requester_routes=dict(type='list'),
+            vpc_id=dict(),
+            vpc_peer_id=dict(),
+            vpc_peering_id=dict(),
+            peer_owner_id=dict(),
+            accept_peer=dict(type='bool', default=False),
+            profile=dict(),
+            accept_with_profile=dict(),
+            resource_tags=dict(type='dict'),
+            state=dict(
+                default='present', choices=[
+                    'present', 'absent', 'accept', 'reject'
+                ]
+            )
+        )
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 is required.')
+
+    check_mode = module.check_mode
+    accept_with_profile = module.params.get('accept_with_profile')
+    accepter_routes = module.params.get('accepter_routes')
+    requester_routes = module.params.get('requester_routes')
+    boto_profile = module.params.get('profile')
+    vpc_id = module.params.get('vpc_id')
+    vpc_peer_id = module.params.get('vpc_peer_id')
+    vpc_peering_id = module.params.get('vpc_peering_id')
+    peer_owner_id = module.params.get('peer_owner_id')
+    accept_peer = module.params.get('accept_peer')
+    tags = module.params.get('resource_tags')
+    state = module.params.get('state').lower()
+
+    if tags:
+        tags = make_tags_in_aws_format(tags)
+
+    if state == 'present' and not tags:
+        err_msg = "parameters state=present and tags are required together"
```

Comment:
Use ansible function `required_if` as part of AnsibleModule object rather than custom statement